code/datums/antagonists/antag_datum.dm

```diff
@@ -1,87 +1,79 @@
+	var/list/typecache_datum_blacklist = list()	//List of datums this type can't coexist with
 
 /datum/antagonist/New()
-	if(!prevented_antag_datum_type)
-		prevented_antag_datum_type = type
-
-/datum/antagonist/Destroy()
-	owner = null
-	return ..()
-
-/datum/antagonist/proc/can_be_owned(mob/living/new_body)
-	return new_body && !new_body.has_antag_datum(prevented_antag_datum_type, TRUE)
-
-/datum/antagonist/proc/give_to_body(mob/living/new_body) //tries to give an antag datum to a mob. cancels out if it can't be owned by the new body
-	if(new_body && can_be_owned(new_body))
-		new_body.antag_datums += src
-		owner = new_body
-		on_gain()
-		. = src //return the datum if successful
-	else
-		qdel(src)
-		. = FALSE
-
-/datum/antagonist/proc/on_gain() //on initial gain of antag datum, do this. should only be called once per datum
-	apply_innate_effects()
-	if(!silent_update && some_flufftext)
-		owner << some_flufftext
+	. = ..()
+	typecache_datum_blacklist = typecacheof(typecache_datum_blacklist)
+	if(ticker)
+		ticker.antagonists[name] += src
+
+/datum/antagonist/proc/apply_innate_effects()	//This handles the application of antag huds/special abilities
+	return
+
+/datum/antagonist/proc/remove_innate_effects()	//This handles the removal of antag huds/special abilities
+	return
 
-/datum/antagonist/proc/apply_innate_effects() //applies innate effects to the owner, may be called multiple times due to mind transferral, but should only be called once per mob
-	//antag huds would go here if antag huds were less completely unworkable as-is
+/datum/antagonist/proc/on_gain() 				//Proc called when the datum is given to a mind. Should only be called once per mind
+	if(owner && owner.current)
+		if(!silent)
+			greet()
+		if(give_special_equipment)
+			give_equipment()
+		if(remove_clown_mut && owner.assigned_role == "Clown")
+			var/mob/living/carbon/human/H = owner.current
+			H << "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself."
+			H.dna.remove_mutation(CLOWNMUT)
 
-/datum/antagonist/proc/remove_innate_effects() //removes innate effects from the owner, may be called multiple times due to mind transferral, but should only be called once per mob
-	//also antag huds but see above antag huds a shit
+	apply_innate_effects()
 
-/datum/antagonist/proc/on_remove() //totally removes the antag datum from the owner; can only be called once per owner
+/datum/antagonist/proc/on_removal()
 	remove_innate_effects()
-	owner.antag_datums -= src
+	if(owner)
+		owner.antag_datums -= src
+		if(!silent && owner.current)
+			farewell()
 	qdel(src)
 
-/datum/antagonist/proc/transfer_to_new_body(mob/living/new_body)
-	remove_innate_effects()
-	if(!islist(new_body.antag_datums))
-		new_body.antag_datums = list()
-	new_body.antag_datums += src
-	owner.antag_datums -= src
-	owner = new_body
-	apply_innate_effects()
+/datum/antagonist/proc/greet()
+	if(text_on_gain)
+		owner.current << text_on_gain
```

Comment:
I'll probably change this to be an actual function and drop the "text_on_gain" var, since some antags have multiple lines/different messages for different mob types.